src/eventDispatchers/mouseEventHandlers/mouseDown.js

```diff
@@ -88,7 +88,22 @@ export default function(evt) {
 
   if (annotationToolsWithMoveableHandles.length > 0) {
     const firstToolWithMoveableHandles = annotationToolsWithMoveableHandles[0];
-    const toolState = getToolState(element, firstToolWithMoveableHandles.name);
+    let toolState = getToolState(element, firstToolWithMoveableHandles.name);
+
+    // create a deep copy of toolState to avoid bad renders of contours after right clicks
+    toolState = JSON.parse(JSON.stringify(toolState));
+
+    const activeRoiId = firstToolWithMoveableHandles.roi.id;
```

Comment:
It's possible that `roi` will be `undefined`, so be sure to handle this case (i.e. only do the new logic when its not null)

Reply:
Fixed